### PR TITLE
JV-oop advanced-hw-solution

### PR DESCRIPTION
**Pull request by JohnGrey17**

### CODE REVIEW

File: src/main/java/core/basesyntax/figures/Circle.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    @Override
+    public void draw() {
+        System.out.println("Figure: " + getClass().getSimpleName() + " radius: " + getRadius()
+                + ", color: " + getColor());
+    }
+
+    @Override
+        public double getArea() {
+        return Math.PI * radius * radius;
+    }
```
Comment (Rommelua) [inline src/main/java/core/basesyntax/figures/Circle.java:23]:
```suggestion
    public double getArea() {
        return Math.PI * radius * radius;
    }

```

File: src/main/java/core/basesyntax/figures/Figure.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+    @Override
+    public void draw() {
+
+    }
+
+    @Override
+    public double getArea() {
+        return 0;
+    }
```
Comment (Rommelua) [inline src/main/java/core/basesyntax/figures/Figure.java:25]:
```suggestion
```

File: src/main/java/core/basesyntax/figures/Figure.java [hunk 3]
```diff
@@ -0,0 +1,26 @@
+package core.basesyntax.figures;
+
+import core.basesyntax.AreaCalculator;
+import core.basesyntax.Drawable;
+
+public class Figure implements Drawable, AreaCalculator {
```
Comment (Rommelua) [inline src/main/java/core/basesyntax/figures/Figure.java:6]: abstract

File: src/main/java/core/basesyntax/supplier/ColorSupplier.java [hunk 4]
```diff
@@ -0,0 +1,11 @@
+package core.basesyntax.supplier;
+
+import java.util.Random;
+
+public class ColorSupplier {
+    public String getRandomColor() {
+        int index = new Random().nextInt(Color.values().length);
```
Comment (Rommelua) [inline src/main/java/core/basesyntax/supplier/ColorSupplier.java:7]: image

File: src/main/java/core/basesyntax/supplier/Figure.java [hunk 5]
```diff
@@ -0,0 +1,9 @@
+package core.basesyntax.supplier;
+
+public enum Figure {
```
Comment (Rommelua) [inline src/main/java/core/basesyntax/supplier/Figure.java:3]: what is it for?

File: src/main/java/core/basesyntax/supplier/FigureSupplier.java [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+import core.basesyntax.figures.RightTriangle;
+import core.basesyntax.figures.Square;
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int FIGURE_COUNT = 6;
+    private Random random = new Random();
+    private ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Circle getCircle() {
```
Comment (Rommelua) [inline src/main/java/core/basesyntax/supplier/FigureSupplier.java:16]: private
Private ones (methods, classes) should be after public ones.

File: src/main/java/core/basesyntax/supplier/FigureSupplier.java [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+            default:
+                return getDefaultFigure();
+        }
+
+    }
+
+    public Figure getDefaultFigure() {
+        Circle circle = new Circle();
+        circle.setColor(Color.WHITE.name());
+        circle.setRadius(10);
```
Comment (Rommelua) [inline src/main/java/core/basesyntax/supplier/FigureSupplier.java:78]: image

File: src/main/java/core/basesyntax/supplier/FigureSupplier.java [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+        isoscelesTrapezoid.setHigh(random.nextInt(FIGURE_COUNT));
+        isoscelesTrapezoid.setFirstSide(random.nextInt(FIGURE_COUNT));
+        isoscelesTrapezoid.setSecondSide(random.nextInt(FIGURE_COUNT));
+        isoscelesTrapezoid.setColor(colorSupplier.getRandomColor());
+        return isoscelesTrapezoid;
+    }
+
+    public Figure getRandomFigure() {
+        int index = random.nextInt(FIGURE_COUNT);
+        Figure figure = new Figure();
```
Comment (Rommelua) [inline src/main/java/core/basesyntax/supplier/FigureSupplier.java:57]:
```suggestion
```

**Review by Rommelua — CHANGES_REQUESTED** [review]

**Comment by JohnGrey17** [comment]
close

**Pull request closed without merging**